lib/reline.rb

```diff
@@ -337,30 +340,24 @@ def readline(prompt = '', add_hist = false)
         io_gate.set_default_key_bindings(config)
       end
 
+      line_editor.print_nomultiline_prompt(prompt)
+      line_editor.update_dialogs
```

Comment:
Please let me know for study purposes. Why do I need to update dialogs before `read_io`?

Reply:
Reline has an ability to always show dialog. Reline needs to setup and render dialog on initial render, before first read_io.
```
irb(main):001> Reline.add_dialog_proc(:FOO,->{Reline::DialogRenderInfo.new contents:['DIALOG'], pos: cursor_pos})
irb(main):002> 
               DIALOG
```
This is tested in `test_dialog_narrower_than_screen` and `test_dialog_with_fullwidth_scrollbar`